src/librustdoc/clean/mod.rs

```diff
@@ -1774,33 +1724,25 @@ impl<'tcx> Clean<Constant> for ty::Const<'tcx> {
 
 impl Clean<Item> for hir::StructField<'_> {
     fn clean(&self, cx: &DocContext<'_>) -> Item {
-        let local_did = cx.tcx.hir().local_def_id(self.hir_id);
-
-        Item {
-            name: Some(self.ident.name).clean(cx),
-            attrs: self.attrs.clean(cx),
-            source: self.span.clean(cx),
-            visibility: self.vis.clean(cx),
-            stability: get_stability(cx, local_did.to_def_id()),
-            deprecation: get_deprecation(cx, local_did.to_def_id()),
-            def_id: local_did.to_def_id(),
-            inner: StructFieldItem(self.ty.clean(cx)),
-        }
+        let what_rustc_thinks = Item::from_hir_id_and_parts(
+            self.hir_id,
+            Some(self.ident.name),
+            StructFieldItem(self.ty.clean(cx)),
+            cx,
+        );
+        Item { visibility: self.vis.clean(cx), ..what_rustc_thinks }
```

Comment:
Why does rustdoc override visibility here?

Reply:
It turns out this is actually necessary: if you have

```rust
pub enum Foo {
    Bar {
        qux: (),
    }
}
```
rustdoc will render the `Bar` as `pub(crate) Bar` unless you change the visibility: https://github.com/rust-lang/rust/pull/77820/checks?check_run_id=1400308817#step:24:1755.

I added a comment and changed it back to the old behavior.